src/parser.ts

```diff
@@ -187,9 +183,197 @@ function getFunctionName(func: ts.FunctionLikeDeclaration): string {
             }
         }
     }
+    return appendNecessaryPrefix(nameText, func);
+}
+
+function appendNecessaryPrefix(
```

Comment:
This function is doing two things: it is determining the prefix and performing concatenation.

Can we make it just determine the prefix (call it `getPrefix`) so the caller can do the concatenation?